src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -62,6 +66,91 @@ constexpr int8_t kDefaultDeadBand                 = 25; // 2.5C is the default
 
 #define FEATURE_MAP_DEFAULT FEATURE_MAP_HEAT | FEATURE_MAP_COOL | FEATURE_MAP_AUTO
 
+class ThermostatAttrAccess : public AttributeAccessInterface
+{
+public:
+    ThermostatAttrAccess() : AttributeAccessInterface(Optional<EndpointId>::Missing(), Thermostat::Id) {}
+
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+    CHIP_ERROR Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder) override;
+};
+
+ThermostatAttrAccess gThermostatAttrAccess;
+
+CHIP_ERROR ThermostatAttrAccess::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    VerifyOrDie(aPath.mClusterId == Thermostat::Id);
+
+    bool LTNESupported = false;
+    uint32_t OurFeatureMap;
+
+    if (FeatureMap::Get(aPath.mEndpointId, &OurFeatureMap) == EMBER_ZCL_STATUS_SUCCESS &&
+        OurFeatureMap & to_underlying(Feature::kLocalTemperatureNotExposed))
+        LTNESupported = true;
+
+    switch (aPath.mAttributeId)
+    {
+    case LocalTemperature::Id: {
+        if (LTNESupported)
+            return aEncoder.EncodeNull();
+        break;
+    }
+    case RemoteSensing::Id: {
+        if (LTNESupported)
+        {
+            uint8_t valueRemoteSensing;
+            EmberAfStatus status = RemoteSensing::Get(aPath.mEndpointId, &valueRemoteSensing);
+            if (status != EMBER_ZCL_STATUS_SUCCESS)
+            {
+                StatusIB statusIB(ToInteractionModelStatus(status));
+                return statusIB.ToChipError();
+            }
+            valueRemoteSensing &= 0xFE; // clear bit 1 (LocalTemperature RemoteSensing bit)
+            return aEncoder.Encode(valueRemoteSensing);
+        }
+        break;
```

Comment:
is it ok to just return CHIP_NO_ERROR if LTNE is not supported, without encoding anything?

Reply:
Yes.  That will fall back to just reading from the attribute store.

Or we could encode the value ourselves, since we have it, and just mask off the bit if LTNESupported.